src/CardVerifier.sol

```diff
@@ -0,0 +1,213 @@
+        
+        // Verify first hole card (position 0)
+        bool card1Valid = Bn254.verifyPartialDecrypt(
+            bDeckSigned[0],
+            card1Opener,
+            pkB
+        );
+        
+        // Verify second hole card (position 1)
+        bool card2Valid = Bn254.verifyPartialDecrypt(
+            bDeckSigned[1],
+            card2Opener,
+            pkB
+        );
+        
+        return card1Valid && card2Valid;
+    }
+
+    /// @notice Verify hole cards for player B (third and fourth cards in deck)
+    /// @dev Verifies cards at positions 2 and 3 using BN254 pairing.
+    /// Player A must provide decryptions to reveal Player B's hole cards.
+    /// Checks: e(bDeckSigned[i], pkA) == e(cardOpener[i], G2_BASE)
+    /// @param pkA G2 public key of player A (128 bytes)
+    /// @param bDeckSigned Final deck encrypted by both players (array of G1 points, each 64 bytes)
+    /// @param card1Opener G1 point representing third card decrypted by A (64 bytes)
+    /// @param card2Opener G1 point representing fourth card decrypted by A (64 bytes)
+    /// @return True if both cards are correctly decrypted
+    function verifyHoleB(
+        bytes memory pkA,
+        bytes[] memory bDeckSigned,
+        bytes memory card1Opener,
+        bytes memory card2Opener
+    ) internal view returns (bool) {
+        require(bDeckSigned.length >= 4, "Deck must have at least 4 cards");
+        require(card1Opener.length == 64, "card1Opener must be 64 bytes");
+        require(card2Opener.length == 64, "card2Opener must be 64 bytes");
+        
+        // Verify first hole card for B (position 2)
+        bool card1Valid = Bn254.verifyPartialDecrypt(
+            bDeckSigned[2],
+            card1Opener,
+            pkA
+        );
+        
+        // Verify second hole card for B (position 3)
+        bool card2Valid = Bn254.verifyPartialDecrypt(
+            bDeckSigned[3],
+            card2Opener,
+            pkA
+        );
+        
+        return card1Valid && card2Valid;
+    }
+
+    /// @notice Verify flop cards (cards 5, 6, 7 in deck - indices 4, 5, 6)
+    /// @dev Verifies three flop cards using both players' public keys.
+    /// Both players must provide decryptions for public cards.
+    /// Checks: e(bDeckSigned[i], pkA) == e(cardAOpeners[i], G2_BASE) AND
+    ///         e(cardAOpeners[i], pkB) == e(cardBOpeners[i], G2_BASE)
+    /// @param pkA G2 public key of player A (128 bytes)
+    /// @param pkB G2 public key of player B (128 bytes)
+    /// @param bDeckSigned Final deck encrypted by both players (array of G1 points, each 64 bytes)
+    /// @param cardAOpeners Array of 3 G1 points representing cards partially decrypted by A (each 64 bytes)
+    /// @param cardBOpeners Array of 3 G1 points representing cards fully decrypted by B (each 64 bytes)
+    /// @return True if all three flop cards are correctly decrypted
+    function verifyFlop(
```

Comment:
Flop, Turn and River checks are pretty much identical. Let's make one function instead: `verifyPublic(pkA, pkB, bDeckSigned, cardAOpener, cardBOpener, cardIndex)`. Replace all 3 functions with new 1 function and update docs and tests.